backend/apps/api/v1/models.py

```diff
@@ -717,7 +725,55 @@ def clean(self) -> None:
             raise ValidationError("Entity's category is not in category.slug = `datetime`.")
         return super().clean()
 
+class Poll(BaseModel):
+    id = models.UUIDField(primary_key=True, default=uuid4)
+    entity = models.ForeignKey("Entity", on_delete=models.CASCADE, related_name="polls")
+    frequency = models.IntegerField()
+    lag = models.IntegerField(blank=True, null=True)
+    latest = models.DateTimeField(blank=True, null=True)
+    raw_data_source = models.ForeignKey(
+        "RawDataSource",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="polls",
+    )
+    information_request = models.ForeignKey(
+        "InformationRequest",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="polls",
+    )
+
+    graphql_nested_filter_fields_whitelist = ["id"]
+
+    def __str__(self):
+        return f"{str(self.frequency)} {str(self.entity)}"
+
+    class Meta:
+        """Meta definition for Poll."""
+
+        db_table = "poll"
+        verbose_name = "Poll"
+        verbose_name_plural = "Polls"
+        ordering = ["frequency"]
 
+    def clean(self) -> None:
+        """Assert that only one of "raw_data_source", "information_request" is set"""
+        count = 0
+        if self.raw_data_source:
+            count += 1
+        if self.information_request:
+            count += 1
+        if count != 1:
+            raise ValidationError(
+                "One and only one of 'raw_data_source',"
+                " or 'information_request' must be set."
+            )
```

Comment:
Remover essa lógica de count parar simplificar o código. Podemos usar um xor aqui com o "bool" e o "=="

```suggestion
        if  bool(self.raw_data_source) == bool(self.information_request):
            raise ValidationError(
                "One and only one of 'raw_data_source',"
                " or 'information_request' must be set."
            )
```

Se ambos os campos raw_data_source e information_request estiverem presentes no modelo, o erro será acionado. No entanto, se ambos estiverem ausentes, ou seja, se não houver nem raw_data_source nem information_request, um erro também será disparado. Agora, se um dos campos estiver presente e o outro não, nenhum erro será disparado.